### PR TITLE
Fix for 'api-basket-add-voucher', not working with restframework 3

### PR DESCRIPTION
**Pull request by maerteijn**
Also adjusted travis.yml so we test against Django 1.8

### CODE REVIEW

**Comment by specialunderwear** [comment]
Do es xml fixture

File: oscarapi/views/basket.py [hunk 1]
```diff
@@ -126,7 +127,11 @@ def add_voucher(request, format=None):
                                              context={'request': request})
     if v_ser.is_valid():
         basket = get_basket(request)
-        voucher = v_ser.object
```
Comment (specialunderwear) [inline oscarapi/views/basket.py]: Did that become validated_data? But how do you get a model instance out of a modelserializet in rf3???

Is there really not a more direct way?

Comment (specialunderwear) [inline oscarapi/views/basket.py]: V_ser.instance?

Comment (maerteijn) [inline oscarapi/views/basket.py]: This is a n API change how this works in the restframework, see: http://www.django-rest-framework.org/topics/3.0-announcement/#serializers and then especially:

```Validating the data would create an object instance. This instance would be available as serializer.object.```

We used this mechanism to retrieve the object itself, even if it was not saved (for example in the login serializer). However, with the add-voucher call we don't want the object to be created or upated (which is now supported in the .create and .update methods of a serializer) as we just want to check if the voucher exists and if it's valid.

So we could set the .instance property in the serializer to the validated correct voucher (self.instance = voucher), or do it like this.

**Comment by specialunderwear** [comment]
Tomorrow I will look and merge

**Comment by maerteijn** [comment]
@specialunderwear See also the alternative implementationusing the .instance property: https://github.com/django-oscar/django-oscar-api/pull/46

(also fixed for the login serializer there)

**Comment by specialunderwear** [comment]
I will take the other one, thnx.

**Pull request closed without merging**